dev/tests/api-functional/testsuite/Magento/GraphQl/Quote/Customer/GetAvailableShippingMethodsTest.php

```diff
@@ -0,0 +1,190 @@
+
+        $expectedAddressData = [
+            'amount' => 10,
+            'base_amount' => 10,
+            'carrier_code' => 'flatrate',
+            'carrier_title' => 'Flat Rate',
+            'error_message' => '',
+            'method_code' => 'flatrate',
+            'method_title' => 'Fixed',
+            'price_incl_tax' => 10,
+            'price_excl_tax' => 10,
+        ];
+        self::assertEquals(
+            $expectedAddressData,
+            $response['cart']['shipping_addresses'][0]['available_shipping_methods'][0]
+        );
+    }
+
+    /**
+     * Test case: get available shipping methods from quote of another customer
+     *
+     * @magentoApiDataFixture Magento/Customer/_files/three_customers.php
+     * @magentoApiDataFixture Magento/Checkout/_files/quote_with_address_saved.php
+     */
+    public function testGetAvailableShippingMethodsFromAnotherCustomerCart()
+    {
+        $maskedQuoteId = $this->getMaskedQuoteIdByReservedOrderId('test_order_1');
+
+        $this->expectExceptionMessage(
+            "The current user cannot perform operations on cart \"$maskedQuoteId\""
+        );
+
+        $this->graphQlQuery(
+            $this->getQuery($maskedQuoteId),
+            [],
+            '',
+            $this->getHeaderMap('customer2@search.example.com')
+        );
+    }
+
+    /**
+     * Test case: get available shipping methods when all shipping methods are disabled
+     *
+     * @magentoApiDataFixture Magento/Checkout/_files/quote_with_address_saved.php
+     * @magentoApiDataFixture Magento/OfflineShipping/_files/disable_offline_shipping_methods.php
+     */
+    public function testGetAvailableShippingMethodsIfAllShippingsAreDisabled()
+    {
+        $maskedQuoteId = $this->getMaskedQuoteIdByReservedOrderId('test_order_1');
+        $response = $this->graphQlQuery($this->getQuery($maskedQuoteId), [], '', $this->getHeaderMap());
+
+        self::assertEquals(0, count($response['cart']['shipping_addresses'][0]['available_shipping_methods']));
+    }
+
+    /**
+     * Test case: get available shipping methods from non-existent cart
+     *
+     * @magentoApiDataFixture Magento/Customer/_files/customer.php
+     * @expectedException \Exception
+     * @expectedExceptionMessage Could not find a cart with ID "non_existent_masked_id"
+     */
+    public function testGetAvailableShippingMethodsOfNonExistentCart()
+    {
+        $maskedQuoteId = 'non_existent_masked_id';
+        $query = $this->getQuery($maskedQuoteId);
+
+        $this->graphQlQuery($query, [], '', $this->getHeaderMap());
+    }
+
+    /**
+     * @param string $maskedQuoteId
+     * @return string
+     */
+    private function getQuery(
+        string $maskedQuoteId
+    ): string
+    {
+        return <<<QUERY
+query {
+  cart (cart_id: "{$maskedQuoteId}") {
+    shipping_addresses {
+        available_shipping_methods {
+          amount
+          base_amount
```

Comment:
Please, check scheme
Looks like we don't have these fields in scheme declaration

Reply:
> Looks like we don't have these fields in scheme declaration
@naydav , could you please check `app/code/Magento/QuoteGraphQl/etc/schema.graphqls, lines` 197-206.